codemp/rd-rend2/tr_glsl.cpp

```diff
@@ -374,6 +376,20 @@ static size_t GLSL_GetShaderHeader(
 						fbufWidthScale,
 						fbufHeightScale));
 
+	if (r_cubeMapping->integer)
+	{
+		int cubeMipSize = CUBE_MAP_SIZE; // r_cubemapSize->integer;
+		int numRoughnessMips = 0;
+
+		while (cubeMipSize)
+		{
+			cubeMipSize >>= 1;
+			numRoughnessMips++;
+		}
+		numRoughnessMips = MAX(1, numRoughnessMips - 2);
```

Comment:
Perhaps move this code to calculate the number of mips into a function? The same code appears in another one of your changes too.